src/Badges.sol

```diff
@@ -308,7 +308,44 @@ contract Badges is
       "mintWithConsent: invalid recipient signature"
     );
 
-    mint(_recipient, _uri, raftTokenId);
+    return mint(_recipient, _uri, raftTokenId);
+  }
+
+  function merkleMintWithConsent(
+    address _recipient,
+    address _issuer,
+    string calldata _uri,
+    bytes calldata _issuerSignature,
```

Comment:
I'm also wondering if we can just generally recover the signers from signatures, why do we need to ask for explicit addresses for issuer and claimant, esp since these will can be called by anyone.. can we think through this in detail, please?

Reply:
to recover an issuer's signature in the merkleMintWithConsent scenario, we have to pass both the `_issuerAddress` and `_issuerSignature` into `safeCheckMerkleAgreement` (which recovers the sig and verifies the merkle)

But for the recipient we just do the equivalent of `safeCheckAgreement` which only verified the sig.

Since the sigs need to be treated differently, I'm not sure if this is generalizable

Reply:
you can recover signer without verifying 
https://docs.openzeppelin.com/contracts/2.x/api/cryptography#ECDSA-recover-bytes32-bytes-

But would that be useful though? I'm wondering if we absolutely need to "verify" the signer... wdyt?